### ISSUE TITLE
Expired Reverse Proxy Certificate Cause Cluster Failure

### ISSUE THREAD

**Issue by esbenbach** [microsoft/service-fabric#1007]
Runtime: 7.1.409 

Environment: Azure

OS: Windows 2016

Description: 
We are using the Service Fabric Reverse Proxy over https, using the same certificate as the cluster.
The cluster certificate recently expired, but we had replaced it with a new one (using common name), and everything SEEMED to be fine.

However, we forgot to replace the reverseProxyCertificate thumbprint with the thumbprint of the new certificate - after an automatic image upgrade, the seed nodes started to fail.
I have spent the better part of a day recovering from this error with MS Support, and during that time we even managed to break the application because the certifcate thumbprint is also in our application settings file, and the engineer told us to do something that changed the cert that the reverse proxy uses.

So basically I have few questions/issues:
1. Why can't the cluster operate with an invalid/expired Reverse Proxy Certificate?
2. Why can't I configure the reverse proxy to use commonName so I don't have to worry about it.
3. Why is there no visible alert in SFX to tell me the certificate that the proxy is using is going to expire (the cluster certificate warning is very much in your face, i would expect the proxy to be the same).

---
Assignees: @microsoft/service-fabric-triage
